examples/borderless.rs

```diff
@@ -0,0 +1,42 @@
+use dioxus::prelude::*;
+use dioxus_desktop::desktop_context::DesktopContext;
+
+fn main() {
+    dioxus::desktop::launch_cfg(app, |cfg| {
+        cfg.with_window(|w| w.with_title("BorderLess Demo").with_decorations(false))
+    });
+}
+
+fn app(cx: Scope) -> Element {
+    let desktop = cx.consume_context::<DesktopContext>().unwrap();
+
+    let drag = desktop.clone();
+    let close = desktop.clone();
```

Comment:
It would be better if we had a hook called "use_desktop_context()"

```rust
fn use_desktop_context(cx: &ScopeState) -> &Rc<DesktopContext> {
    cx.use_hook(|_| cx.consume_context::<DesktopContext>()).as_ref().unwrap()
}
```

That way desktop context comes out as an &Rc which can be borrowed into the callbacks without issue.